Apply new button style guide

ArturDolzan opened the pull request:
### Description

This PR implements the new button style guide, adjusting variants, sizes, and paddings.

- Updated Button and ButtonLink components to follow the new style guide.
- Introduced small, xSmall, and xLarge size options with proper spacing and typography.
- Simplified the internal logic for computing classes and added a utility to merge Tailwind classes with conflict resolution.
- Adjusted button usage across the app, especially inside drawers, where smaller buttons (xSmall, small) are now applied for better visual balance.

### Screenshots

https://github.com/user-attachments/assets/59947b5c-6676-4f14-b120-491148925c78

### Related issue(s)

Closes #1376
Closes #1389
Related to #1334  

### Checklist

- [x] Manual testing passed.
- [x] Automated tests added, or N/A.
- [ ] Documentation updated, or N/A.
- [ ] Environment variables set in CI, or N/A.

gabmontes reviewed (changes requested):
@ArturDolzan conflicts!

gabmontes commented on portal/utils/simpleMergeTailwindClasses.ts:
```diff
@@ -0,0 +1,39 @@
+ * @param deduplicationPrefixes - An optional array of prefixes to use for deduplication (defaults to ['px-', 'py-']).
+ * @returns A single string containing the merged and deduplicated Tailwind CSS classes.
+ */
+export function simpleMergeTailwindClasses({
```
How is this related to [tailwind-merge](https://www.npmjs.com/package/tailwind-merge)? Should we use that library instead, as is a more complete merging solution?

gndelia replied:
I know it's not _exactly_ the same, but in general, for these components that have a very defined set of options (Variants), I'd prefer avoiding exposing the `className` to the caller, because if we would, the consumer could potentially override any option... I think we shouldn't allow that, as it does not favor the consistency of the components across the app. If an option becomes so important, maybe it needs to be part of the props.

ArturDolzan replied:
@gndelia I agree that avoiding className exposure helps enforce consistency.. there’s just one edge case still open: we have some buttons (actually just two) that render only a single icon, like  the reload tx history, with no label or additional content. For those, the padding is different (px-1.5 -> 6px).

In this case, removing the className exposure, I was thinking of introducing a prop like isIconOnly to handle that scenario explicitly. Alternatively, I tried to infer it internally (e.g., checking if children is a single valid React element without children), but it’s not very reliable..

Do you think isIconOnly as a prop is acceptable in this case? Or do you have a better idea on how we could detect and handle this padding variation without breaking the style contract?

gndelia replied:
yeah, I understand what you're saying. For that reason I made (and exists). the `ButtonLink` component, where it looks like a button, but it's actually not a button (it's a link!)

Maybe you can use a `ButtonIcon` component, and it does something like this

```tsx
const (children, buttonProps) => (

<div className="[&>button]:px-1.5>
  <Button {...buttonProps}>{children}</Button>
/>
```

it's not that far from `isIconOnly`, that's correct, but it keeps the `Button` API simpler.

_actually_, I think that rather than setting `px-.1.5`, you need to set `px-0` (Removing it) and make the height and width match.. So basically I think this component would use its own set of sizeMaps:



```tsx

const (children, size, ...buttonProps) => (

<div className="[&>button]:px-1.5>
  <Button {...buttonProps}>{children}</Button>
/>
```

gabmontes commented on portal/components/button.tsx:
```diff
@@ -1,65 +1,80 @@
-    focus:shadow-button-primary-focused`,
-  secondary: `text-neutral-950 bg-white border-neutral-300/55 hover:bg-neutral-100
-    disabled:bg-neutral-100 shadow-button-secondary focus:shadow-button-secondary-focused`,
+  primary: `primary-button border-button-primary bg-button-primary text-white 
+  transition-all duration-200 hover:bg-button-primary-hovered 
+  disabled:bg-button-primary-disabled disabled:shadow-button-primary-disabled shadow-button-primary 
+  focus:shadow-button-primary-focused border border-solid`,
+  secondary: `text-neutral-950 bg-white hover:bg-neutral-50 disabled:bg-neutral-50 
+  shadow-button-secondary focus:shadow-button-secondary-focused disabled:shadow-button-secondary-disabled`,
+  tertiary: `text-neutral-950 bg-transparent hover:bg-neutral-100 focus:bg-neutral-100 focus:shadow-button-tertiary-focused`,
```
You may want to play with the indentation here to make this more readable. Or just use plain single quotes. Same for `commonCss` above.

ArturDolzan replied:
Improved here 5370e454620d9b73d863e7bb6179f0e9401e9069

dvnahuel commented:
Can we add a transition so the hovering interactions feels more smooth? let's try something like `transition: all 0.3s ease;
`

dvnahuel commented:
CTA's focus state should trigger only on keyboard focus, not mouse click.

ArturDolzan commented on portal/components/button.tsx:
```diff
@@ -1,65 +1,80 @@
-type ButtonProps = ComponentProps<'button'> & Variant & FontSize & Height
+export type ButtonSize = 'xSmall' | 'small' | 'xLarge'
+
+const sizeMap: Record<ButtonSize, string> = {
+  small: 'h-8 text-sm gap-x-2 px-3',
+  xLarge: 'h-11 text-mid gap-x-2 px-4',
+  xSmall: 'h-7 text-xs gap-x-1 px-2.5',
+} as const
```
Well, it was discussed with @dvnahuel and he mentioned that it was not needed to create, for example, the Large one for now, but it could be done in the future.. so to follow his definitions [here](https://github.com/hemilabs/ui-monorepo/issues/1389), I used basically the same nomenclature. 

Please let me know if you guys prefer another nomenclature.. I'm good with both.

gndelia replied:
no, it's ok, but I got curious 😅

dvnahuel commented:
XS style CTA's have should have 6px border radius.

dvnahuel commented:
Reload CTA here should be styles as 'secondary'
<img width="107" height="121" alt="Screenshot 2025-07-25 at 12 25 31 PM" src="https://github.com/user-attachments/assets/d08b5514-f7d6-48cc-bf64-1a73d45246fb" />

dvnahuel commented:
The menu CTA on mobile and tablet views should also be styled as secondary.
<img width="708" height="222" alt="Screenshot 2025-07-25 at 12 29 06 PM" src="https://github.com/user-attachments/assets/27ea1f60-d440-4382-946e-8a10eecb1a93" />

gndelia commented:
> Reload CTA here should be styles as 'secondary' <img alt="Screenshot 2025-07-25 at 12 25 31 PM" width="107" height="121" src="https://private-user-images.githubusercontent.com/23080631/470860668-d08b5514-f7d6-48cc-bf64-1a73d45246fb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM0NTc1NDIsIm5iZiI6MTc1MzQ1NzI0MiwicGF0aCI6Ii8yMzA4MDYzMS80NzA4NjA2NjgtZDA4YjU1MTQtZjdkNi00OGNjLWJmNjQtMWE3M2Q0NTI0NmZiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzI1VDE1MjcyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhNGZmYWI3ZDY1NDk1ZDQ5OGNjNWM2OTI1MjllYWZkN2M4MjQwYmYzNzIwZWJmNTk3MWE3MzJlNGRjMTYwZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cJ8d_9HWuEnUTLk1a4nXT9IFfPvRlgAEOuwZ786qYfs">

The issue says the other way around! https://github.com/hemilabs/ui-monorepo/issues/1389

<img width="410" height="131" alt="image" src="https://github.com/user-attachments/assets/79c8a31c-3569-4b32-bc3b-b82c193643ff" />

dvnahuel commented:
Something we need to keep in mind is the following: CTAs with icons should use the same colors as shown in Figma.
For example, in the case of the 'Connect Wallets' CTA, the wallet icons should be set (in the default state) to grayscale/500, and on hover, to grayscale/950.
<img width="246" height="212" alt="Screenshot 2025-07-25 at 12 51 29 PM" src="https://github.com/user-attachments/assets/c809a772-f7a1-46b8-a351-ccbaacf2103e" />

ArturDolzan commented on portal/components/button.tsx:
```diff
@@ -1,65 +1,80 @@
-const commonCss = `box-content flex items-center justify-center
-  rounded-lg border border-solid px-3 py-1.5 font-medium disabled:opacity-50`
+const commonCss = `box-border flex items-center justify-center
+  rounded-lg py-1.5 font-medium disabled:opacity-50`
```
Fixed here 5370e454620d9b73d863e7bb6179f0e9401e9069

ArturDolzan commented:
@dvnahuel this commit 5370e454620d9b73d863e7bb6179f0e9401e9069 addresses several feedback points related to CTA styles and behavior:

- Focus state: CTA focus styles now appear only when navigating via keyboard (focus-visible), not on mouse click.
- Primary button shadow: Adjusted shadows to better match the design in Figma.
- XS CTAs: Ensured xSmall buttons have a 6px (rounded-md) border radius.
- Reload CTA: Updated to use the secondary style, both in the corresponding screen and mobile menu.
- Oops screen CTA: Now uses the small primary style as expected.
- Icon colors: Updated icons inside buttons (e.g. Connect Wallet) to match the expected grayscale values: text-neutral-500 by default, text-neutral-950 on hover.

gndelia commented on portal/components/connectWallets/wallets.tsx:
```diff
@@ -58,8 +58,8 @@ const ConnectWalletButton = function ({
       {rightIcon || (
-        <div className="ml-auto">
-          <Chevron.Right />
+        <div className="group-hover ml-auto">
+          <Chevron.Right className="group-hover:[&>path]:fill-neutral-950" />
```
I think this is incorrect. Check out the [group-hover](https://v3.tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state) docs

```suggestion
        <div className="group ml-auto">
          <Chevron.Right className="group-hover:[&>path]:fill-neutral-950" />
```

The parent uses `group` and the children uses `group-hover`. They can also be named, like 


```suggestion
        <div className="group/container ml-auto">
          <Chevron.Right className="group-hover/container:[&>path]:fill-neutral-950" />
```

but apparently that's not mandatory

ArturDolzan replied:
Fixed here 3d67e7a7c9bf96bd276d8319e7c91fe8ccd79f89. Thanks!

dvnahuel commented:
The primary CTA shadows are off
<img width="300" height="112" alt="Screenshot 2025-07-29 at 10 45 16 AM" src="https://github.com/user-attachments/assets/e71f1edf-7bbd-4624-b07e-f99d4f5cd3ca" />

gndelia commented:
@dvnahuel  a minor question, but

<img width="1141" height="352" alt="image" src="https://github.com/user-attachments/assets/028104ef-50ba-4afe-b57d-458dedb1c339" />


Should we update the image with the "Deploy" button, so the button matches our new design? To be fair, I don't think it's even matching our previous design entirely.

(Note this could be done in a separate PR / issue)

dvnahuel commented:
The Connect Wallet icon should use neutral/500 by default, and change to neutral/950 on hover.
<img width="198" height="78" alt="Screenshot 2025-07-29 at 10 56 10 AM" src="https://github.com/user-attachments/assets/f6dc8eeb-fb54-4dd5-bb88-3db64f6bc364" />

ArturDolzan commented on portal/app/[locale]/tunnel/transaction-history/_components/reloadHistory.tsx:
```diff
@@ -29,12 +30,13 @@ export const ReloadHistory = function () {
-      <ReloadIcon className="[&>path]:group-hover/reload-history:fill-neutral-950" />
-    </button>
+      <ReloadIcon />
+    </ButtonIcon>
```
Fixed here 3d67e7a7c9bf96bd276d8319e7c91fe8ccd79f89. Thanks!

dvnahuel commented:
The 'hamburger' CTA should have the same interacion as the secondary CTA (on-hover the background-color and icon color changes to natural-950)
<img width="62" height="53" alt="Screenshot 2025-07-29 at 1 49 17 PM" src="https://github.com/user-attachments/assets/4380b224-c286-46c6-8077-c36b49f02387" />

dvnahuel commented:
> > @dvnahuel a minor question, but
> > <img alt="image" width="1141" height="352" src="https://private-user-images.githubusercontent.com/352474/472025617-028104ef-50ba-4afe-b57d-458dedb1c339.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTM4MTEwMzAsIm5iZiI6MTc1MzgxMDczMCwicGF0aCI6Ii8zNTI0NzQvNDcyMDI1NjE3LTAyODEwNGVmLTUwYmEtNGFmZS1iNTdkLTQ1OGRlZGIxYzMzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNzI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDcyOVQxNzM4NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNDk0YWRlZWQ4MTU1NjQyMTM2YjE1MDNlMzQxZDIxZjkwMzA4YjYwZDU3NDgxZjNiODgwNjEyMzVkNDkwNGFjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.w9FvL0sKqzJiiMUsPEMs2Rz0x6_SWt62sOzmHUWgyjo">
> > Should we update the image with the "Deploy" button, so the button matches our new design? To be fair, I don't think it's even matching our previous design entirely.
> > (Note this could be done in a separate PR / issue)
> 
> Makes sense, let's put it as a new issue

Issue [here](https://github.com/hemilabs/ui-monorepo/issues/1416)

gndelia reviewed (dismissed):
Code-wise, it LGTM. I'll defer to Nahuel to approve the whole UX side (btw, it looks fantastic!)

gabmontes commented on portal/components/button.tsx:
```diff
@@ -4,51 +4,99 @@ import { isRelativeUrl } from 'utils/url'
-}
+/* eslint-disable sort-keys */
+const buttonSizePresets = {
+  xSmall: {
+    icon: 'h-7 text-xs rounded-md min-w-7',
+    regular: 'h-7 text-xs gap-x-1 px-2.5 rounded-md',
+  },
+  small: {
+    icon: 'h-8 text-sm rounded-lg min-w-8',
+    regular: 'h-8 text-sm gap-x-2 px-3 rounded-lg',
+  },
+  xLarge: {
+    icon: 'h-11 text-mid rounded-lg min-w-11',
+    regular: 'h-11 text-mid gap-x-2 px-4 rounded-lg',
```
Why are we jumping from "small" right to "xLarge"?

ArturDolzan replied:
@gabmontes the naming wasn’t based on `small`, `medium`, and `large` because during the initial discussion with @dvnahuel , we aligned on only implementing a specific size for now, and following the naming from his reference to keep it consistent. That said, I’m totally open to renaming them if using a more generic convention like small, medium, large would make things clearer.

Let me know what you prefer..

dvnahuel replied:
This is how I’ve been structuring the designs, mainly because, in the current portal, there isn’t really a place where a ‘large (40px)’ CTA fits naturally. The x-large (44px) button is mostly used in the tunnel form and was almost specifically created for that. Everywhere else across the portal we’re mostly using xSmall or small button sizes.

gabmontes replied:
Got it. Thanks for the explanation.
